orttraining/orttraining/core/session/training_session.cc

```diff
@@ -70,9 +79,14 @@ Status SetupOptimizerParams(
         loss_scale_input_name.has_value() ? loss_scale_input_name.value() : "";
     opt_node_config.use_mixed_precision_moments = optimizer_config.use_mixed_precision_moments;
 
-    const auto mixed_precision_weight_name_it = fp32_weight_names_to_mixed_precision_node_args.find(weight_name);
+    const auto mixed_precision_weight_name_it = fp32_weight_names_to_mixed_precision_node_args.find(w_n);
     if (mixed_precision_weight_name_it != fp32_weight_names_to_mixed_precision_node_args.end()) {
       opt_node_config.mixed_precision_weight_arg = mixed_precision_weight_name_it->second;
+    } else {
```

Comment:
I recalled I fixed some bugs. could you please refer to the pengwa/bart_ort branch for those fixes?

Reply:
this part has been fixed after debugging